### PR TITLE
fix: workaround for lack of tiller input for nose wheel steering (NWS)

### LINKED ISSUE #5851: Workaround for lack of tiller steering
### Aircraft Version

Development

### Description

There is no workable tiller binding in the sim. NWS is programmed with realistic angles according to pedals input. It's requested to override the realistic rudder wheel steering with tiller steering angles otherwise we cannot taxi properly.

### References (optional)

FCTM:

TILLER AND RUDDER PEDALS USE

Pedals control nosewheel steering at low speed (± 6 ° with full pedal deflection). Therefore, on straight taxiways and on shallow turns, the pilot can use the pedals to steer the aircraft, keeping a hand on the tiller. In sharper turns, the pilot must use the tiller.

### Additional info (optional)

Not being able to taxi properly is worse than not having realistic wheel steering angles with pedals input.

### Discord Username (optional)

_No response_

### PR DESCRIPTION
**Pull request by Villason**
Fixes #5851

## Summary of Changes

This PR fixes the nose wheel steering curves vs rudder axis. Before this PR, the curve cut-off to 6 degrees was too step and made taxiing more difficult that needed to be.

This solution is a temporary fix until Asobo implements a separate axis for the tiller.

## Screenshots (if necessary)

NSW sim curves before and after this PR:

image

## References

Real aircraft curves with pedals and tiller:

image

## Additional context

Discord username (if different from GitHub):

## Testing instructions

In the development console, debug the wheel steering output. Max wheel deflection should be 75 degrees up to 20 knots and decrease linearly down to 6 degrees at 80 knots.

autoland: "thorough testing in various situations, especially with crosswinds."

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page

### CODE REVIEW

**Comment by tracernz** [comment]
cc @donstim

**Comment by donstim** [comment]
The issue this pr is linked to states that there is bug introduced by pr #769. There was no bug introduced by that pr. It was a reasoned attempt to provide a customized solution in the absence of a steering tiller axis in MSFS. The issue of no steering tiller axis in MSFS has not changed.
 
PR #769 provides tiller steering angles at low taxi speeds where needed, but reduces the angle at higher speeds in accordance with the use of rudder pedal steering. It is unclear why you need higher steering angles at 40 knots. You should not trying to make short radius turns at such speeds. Can you clarify why you feel this is needed?

**Comment by Villason** [comment]
> 
> 
> The issue this pr is linked to states that there is bug introduced by pr #769. There was no bug introduced by that pr. It was a reasoned attempt to provide a customized solution in the absence of a steering tiller axis in MSFS. The issue of no steering tiller axis in MSFS has not changed.
> 
> PR #769 provides tiller steering angles at low taxi speeds where needed, but reduces the angle at higher speeds in accordance with the use of rudder pedal steering. It is unclear why you need higher steering angles at 40 knots. You should not trying to make short radius turns at such speeds. Can you clarify why you feel this is needed?

With the current steering max-angle curve, we only have 6 degrees of steering at 40 knots, 37.5 degrees at 30 knots, etc. After the PR, we would have 50 degrees at 40 knots, 62.2 degrees at 30 knots, etc. The difference is significant and it properly represents real aircraft steering performance.

My personal problem is while taxiing in large airports, where I need to taxi abnormally slow to properly take curves.

**Comment by donstim** [comment]
I still don't understand why you need 50 degrees at 40 knots or >60 degrees at 30 knots. For a straight-ahead taxi, you should be limiting speed to a maximum of about 30 knots. For a 90-degree turn, you should be at less than 10 knots.

**Comment by Villason** [comment]
> 
> 
> I still don't understand why you need 50 degrees at 40 knots or >60 degrees at 30 knots. For a straight-ahead taxi, you should be limiting speed to a maximum of about 30 knots. For a 90-degree turn, you should be at less than 10 knots.

Please mind the max angles I am requesting are the ones of the real aircraft. I am running to wide while taxing at normal speeds on difficult taxiways.

**Comment by donstim** [comment]
What are you calling normal taxi speeds?  This pr does not change anything at 20 knots and below. Maybe you can share a video showing the issue you are experiencing and how this pr resolves it?  Sorry to be a pain, but I am just not getting it.

**Comment by donstim** [comment]
I did a couple of maximum crosswind takeoffs, and although it seemed a little more squirrely to me, I couldn't be sure. It seemed like at higher speeds it just caused the nose wheel to skid rather than turn when trying to use the higher steering angle. Where are you seeing a benefit?

**Comment by marcman86** [comment]
Hi, be aware that there is skid of the nosewheel at high turn angles. So the aircraft will understeer. Best turns can be achieved at about 60-80% on the controller. Maybe this solves some of the experienced issues.

**Comment by John-UK** [comment]
Are we creating a fix for something that should not occur or if it does result in an excursion incident?
This IRL incident happened with GS of 18kts. A320 making a 90 deg turn. The safety report highlights that taxi speeds of 20 kts straight and less than 10kts for 90 deg turn are in the aircraft operator’s procedures were based on the Flight Crew Manuals produced by the aircraft manufacturer. The nose wheel continued to skid at 14kts leaving the paved taxiway at 13kts.
The incident concluded the flight crew were going too fast.
FFF747DD-C0DD-4601-8247-85DBFDB110A5

**Comment by aguther** [comment]
I want to remind that this PR can have a large influence on Autoland function. Therefore I recommend thorough testing in various situations, especially with crosswinds.

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Changelog
+
+1. [General] Workaround of lack of tiller input for nose wheel steering (NWS) - @Villason (Villason)
```
Comment (MisterChocker) [inline CHANGELOG.md:3]: Please don't create a new changelog, [use the existing one](https://github.com/Villason/a32nx/blob/nws-pedals/tiller-combo/.github/CHANGELOG.md)

```suggestion
```

**Review by MisterChocker — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (Villason) [reply CHANGELOG.md]: Done.

**Review by Benjozork — APPROVED** [review]

**Comment by Villason** [comment]
> 
> 
> I want to remind that this PR can have a large influence on Autoland function. Therefore I recommend thorough testing in various situations, especially with crosswinds.

Will do. Note the max steering angle curve is the same above 80 knots before and after this PR.

**Comment by MisterChocker** [comment]
> > I want to remind that this PR can have a large influence on Autoland function. Therefore I recommend thorough testing in various situations, especially with crosswinds.
> 
> Will do. Note the max steering angle curve is the same above 80 knots before and after this PR.

The autopilot on autoloadings also needs to function below 80 knots.
So these changes need to be tested and the ap still needs to be able to maintain on the centerline.

**Comment by Villason** [comment]
I tested a few heavy crosswind landings and I didn't notice any wrong
behavior.

Needing more eyes for this, guys.

**Comment by John-UK** [comment]
Is this over taken by #6187?

**Comment by crocket63** [comment]
It should if it proves to work properly enough indeed

**Comment by frankkopp** [comment]
Superseded by #6187

**Pull request closed without merging**